kiwi/utils/rpm_database.py

```diff
@@ -81,6 +81,16 @@ def set_macro_from_string(self, setting):
     def write_config(self):
         self.rpmdb_image.write_config()
 
+    def import_signing_key_to_image(self, key):
+        """
+        Import the given signing key to the image rpm database
+        """
+        Command.run(
+            [
+                'rpm', '--root', self.root_dir, '--import', key
```

Comment:
You should probably make sure this knows the dbpath, otherwise it might not work properly in some circumstances...

Reply:
That import statement takes the host _dbpath setup into account which is what I would have expected and which works well together with the bootstrap phase and the move of the rpmdb to the final location according to the used distribution. I have tested this with a distro that has rpm on /var/lib/rpm(Leap42.3), a distro with rpm on /usr/lib/sysimage(Leap15) and a distro with rpm on /var/lib/rpm(Fedora28) in all kinds of vice versa setup... my head is spinning :sweat_smile: 

So if you have a clear indicator to pass --dbpath to the import call let me know but if not I think I'll stick to the default behavior which seems consistent across all distros

last famous words :smile:

Reply:
In mock, host rpm (TW/LP15) could not read the final chroot's installed package list because even with `--root`, it doesn't read the dbpath from the target environment macro. Don't know why, but it doesn't. So it failed until I passed `--dbpath` as well. The target environment had the rpmdb in `/var/lib/rpm`, but of course SUSE distributions use `/usr/lib/sysimage/rpm` now.

Reply:
ok, then let us fix that. Just a sec

Reply:
done